src/Dialog/Dialog.js

```diff
@@ -105,6 +107,8 @@ function getStyles(props, context) {
       transition: open ?
         transitions.easeOut('0ms', 'left', '0ms') :
         transitions.easeOut('0ms', 'left', '450ms'),
+      direction: isRtl ? 'rtl' : 'ltr',
```

Comment:
BTW, this fix only takes care of primary text direction.
For icons to be properly styled, I guess I also need to touch ListItem too?